From 277f3c0f369f80b5d773b793bc98915bed5b299e Mon Sep 17 00:00:00 2001
From: "vijaya.sundaram" <vijaya.sundaram@lge.com>
Date: Mon, 24 Jun 2019 16:01:58 +0530
Subject: [PATCH] Fix device discovery for broadcast role

:Release Notes:
Fix device discovery for broadcast role

:Detailed Notes:
Broadcast advertisement is not discovered in webos devices,
but same is discoverable on other os.  Irrespective
of discoverable property, add devices to discovery list.

:Testing Performed:
Built and tested

:QA Notes:

:Issues Addressed:
[PLAT-80402] Discover BLE devices on Bluez 5.50

Upstream-Status: Inappropriate[webos specific]
---
 src/adapter.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/adapter.c b/src/adapter.c
index cef25616f..b09855aa5 100644
--- a/src/adapter.c
+++ b/src/adapter.c
@@ -6095,11 +6095,10 @@ static void update_found_devices(struct btd_adapter *adapter,
 	dev = btd_adapter_find_device(adapter, bdaddr, bdaddr_type);
 	if (!dev) {
 		/*
-		 * If no client has requested discovery or the device is
-		 * not marked as discoverable, then do not create new
+		 * If no client has requested discovery, then do not create new
 		 * device objects.
 		 */
-		if (!adapter->discovery_list || !discoverable) {
+		if (!adapter->discovery_list) {
 			eir_data_free(&eir_data);
 			return;
 		}
